internal/fuzzbinder/fuzzclaim.go

```diff
@@ -180,3 +181,122 @@ func parseFuzzClaimBytes(statementBytes []byte) (*intoto.Statement, error) {
 
 	return &statement, nil
 }
+
+// generateFuzzClaimSpec generates a fuzzing claim specification using the
+// fuzzing reports of OSS-Fuzz.
+func generateFuzzClaimSpec(date string, revisionDigest intoto.DigestSet, fuzzParameters *FuzzParameters, fuzzTargets []string) (*FuzzClaimSpec, error) {
+	var fuzzClaimSpec FuzzClaimSpec
+	var projectCrashes Crash
+	var projectFuzzEffort FuzzEffort
+	fuzzersCrashes := make(map[string]*Crash)
+	fuzzersFuzzEffort := make(map[string]*FuzzEffort)
+	fuzzersCoverage := make(map[string]*Coverage)
+	hyphenDate := fmt.Sprintf("%s-%s-%s", date[:4], date[4:6], date[6:])
+	//Get fuzzing statistics.
+	projectCoverage, err := GetCoverage(date, fuzzParameters, "perProject", "")
+	if err != nil {
+		return nil, err
+	}
+
+	for _, fuzzTarget := range fuzzTargets {
+		coverage, err := GetCoverage(date, fuzzParameters, "perTarget", fuzzTarget)
+		if err != nil {
+			return nil, err
+		}
+		fuzzEffort, err := GetFuzzEffort(revisionDigest, hyphenDate, fuzzTarget, fuzzParameters)
+		if err != nil {
+			return nil, err
+		}
+		crash, err := GetCrashes(revisionDigest, hyphenDate, fuzzTarget, fuzzParameters)
+		if err != nil {
+			return nil, err
+		}
+
+		fuzzersCrashes[fuzzTarget] = crash
+		fuzzersFuzzEffort[fuzzTarget] = fuzzEffort
+		fuzzersCoverage[fuzzTarget] = coverage
+
+		projectCrashes.Detected = projectCrashes.Detected || crash.Detected
+		projectFuzzEffort.FuzzTimeSeconds += fuzzEffort.FuzzTimeSeconds
+		projectFuzzEffort.NumberFuzzTests += fuzzEffort.NumberFuzzTests
+	}
+	// Generate fuzzing claim specification.
+	fuzzClaimSpec.PerProject = &FuzzStats{
+		BranchCoverage:  projectCoverage.BranchCoverage,
+		LineCoverage:    projectCoverage.LineCoverage,
+		DetectedCrashes: projectCrashes.Detected,
```

Comment:
Are you computing this and the following two values based on perTarget results because they don't exist perProject, or because it is more accurate this way?

Reply:
The perTarget results are also computed using all fuzzers logs (they are not extracted direclty). In the same way, we can't extract the perProject directly from the logs. 

OSS-Fuzz is also computing the fuzzing efforts and showing them in the html reports, but they are aggregated for all the day (even if multiple revisions are used). So, they can be used.